applications/StructuralMechanicsApplication/custom_utilities/node_search_utility.cpp

```diff
@@ -0,0 +1,41 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Manuel Messmer
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "node_search_utility.h"
+
+namespace Kratos {
+
+void NodeSearchUtility::SearchNodesInRadius(
+        NodeType& rNode,
+        double const Radius,
+        ResultNodesContainerType& rResults ) {
+
+    KRATOS_TRY
+
+    ResultNodesContainerType local_results(mMaxNumberOfNodes);
+    std::size_t num_of_results = 0;
+
+    ResultNodesContainerType::iterator results_iterator = local_results.begin();
```

Comment:
```suggestion
    auto results_iterator = local_results.begin();
```

IMO a good opportunity to use `auto`